Java/core/src/main/java/com/amazon/randomcutforest/ITraversable.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+public interface ITraversable {
```

Comment:
Question. It seems strange this class is not generic.

Reply:
We want the traversal methods to do the right thing regardless of the internal representation of a tree. E.g., if you're writing a new Visitor implementation, you shouldn't care if the tree stores points internally or looks them up on the fly during traversal (which is what the compact tree does).